gemrb/plugins/OpenALAudio/OpenALAudio.cpp

```diff
@@ -765,7 +765,7 @@ int OpenALAudioDriver::SetupNewStream(int x, int y, int z,
 	return stream;
 }
 
-tick_t OpenALAudioDriver::QueueAmbient(int stream, const char* sound)
+tick_t OpenALAudioDriver::QueueAmbient(int stream, const ResRef& sound)
```

Comment:
There's now a bad null check a few lines lower. Also, you didn't convert loadSound accordingly yet, so there will be more implict conversions.

Reply:
IIRC `loadSound` needs to be able to take a whole path. implicit conversions won't be allowed by the end of this journey so they will get resolved.

Reply:
If it needs a whole path, it's definitely not getting it now, since `sound` will be cut off.

Reply:
AFICT it doesnt _need_ a whole path, but whole paths are being passed to it, so it must work with them. See `DisplayStringCoreVC`

Reply:
Ah, ok, and QueueAmbient is only getting ResRefs so far. But why remove the null check, now we're just doing more useless work by pushing the problem deeper.

Reply:
For me it's an optimization of legibility, but it's also true this function is short enough for it to not matter much.